packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -352,6 +369,22 @@ function commitLifeCycles(
           // We could update instance props and state here,
           // but instead we rely on them being set during last render.
           // TODO: revisit this when we implement resuming.
+          if (__DEV__) {
+            if (finishedWork.type === finishedWork.elementType) {
+              warning(
+                instance.props === finishedWork.memoizedProps,
```

Comment:
I thought the whole point of your previous PR is that `memoizedProps` does not include the resolved default props, but `instance.props` does?

Reply:
Oh duh, didn't notice the outer check :D